cmd/ci-operator/main.go

```diff
@@ -846,15 +914,27 @@ func (o *options) writeMetadataJSON() error {
 	m.Pod = o.jobSpec.ProwJobID
 	m.WorkNamespace = o.namespace
 
-	data, err := json.MarshalIndent(m, "", "  ")
-	if err != nil {
-		return err
-	}
-	if o.dry {
-		log.Printf("metadata.json:\n%s", string(data))
-		return nil
+	return m
+}
+
+func (o *options) parseCustomMetadata(customProwMetadataFile string) (customMetadata map[string]string, err error) {
+	// Merge custom metadata into resulting JSON
+	if customProwMetadataFile != "" {
```

Comment:
nit: you can check to see if this is true at the call-site and just don't call this function when there's nothing to do